File: go-client/hashes.go
package goredis

// HSet sets the value of a field in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HSet(key, field string, value interface{}, extra ...interface{}) (interface{}, error) {
	cmdArgs := append([]interface{}{"HSET", key, field, value}, extra...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// HGet retrieves the value of a field in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HGet(key, field string) (interface{}, error) {
	return mustGetClient().SendCommand("HGET", key, field)
}

// HDel deletes one or more fields from a hash stored at key.
// It returns the server's response or an error if the command fails.
func HDel(key string, fields ...string) (interface{}, error) {
	cmdArgs := append([]interface{}{"HDEL", key}, toInterfaceSlice(fields)...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// HGetAll retrieves all fields and values of a hash stored at key.
// It returns the server's response or an error if the command fails.
func HGetAll(key string) (interface{}, error) {
	return mustGetClient().SendCommand("HGETALL", key)
}

// HIncrBy increments the integer value of a field in a hash stored at key by the given increment.
// It returns the server's response or an error if the command fails.
func HIncrBy(key, field string, increment int) (interface{}, error) {
	return mustGetClient().SendCommand("HINCRBY", key, field, increment)
}

// HExists checks if a field exists in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HExists(key, field string) (interface{}, error) {
	return mustGetClient().SendCommand("HEXISTS", key, field)
}

// HLen returns the number of fields in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HLen(key string) (interface{}, error) {
	return mustGetClient().SendCommand("HLEN", key)
}

// HKeys retrieves all field names in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HKeys(key string) (interface{}, error) {
	return mustGetClient().SendCommand("HKEYS", key)
}

// HVals retrieves all values in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HVals(key string) (interface{}, error) {
	return mustGetClient().SendCommand("HVALS", key)
}

// HMSet sets multiple fields in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HMSet(key string, mapping map[string]interface{}) (interface{}, error) {
	cmdArgs := []interface{}{"HMSET", key}
	for k, v := range mapping {
		cmdArgs = append(cmdArgs, k, v)
	}
	return mustGetClient().SendCommand(cmdArgs...)
}

// HMGet retrieves the values of multiple fields in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HMGet(key string, fields ...string) (interface{}, error) {
	cmdArgs := append([]interface{}{"HMGET", key}, toInterfaceSlice(fields)...)
	return mustGetClient().SendCommand(cmdArgs...)
}

// HDelAll deletes all fields in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HDelAll(key string) (interface{}, error) {
	return mustGetClient().SendCommand("HDELALL", key)
}

// HExpire sets a timeout on a field in a hash stored at key.
// It returns the server's response or an error if the command fails.
func HExpire(key, field string, seconds int) (interface{}, error) {
	return mustGetClient().SendCommand("HEXPIRE", key, field, seconds)
}

File: python-client/goredis/hashes.py
from .client import get_client

def HSet(key, field, value, *args):
    """
    Set the string value of a hash field.
    If the field already exists, it updates the value.
    """
    return get_client().send_command("HSET", key, field, value, *args)

def HGet(key, field):
    """Get the value of a hash field."""
    return get_client().send_command("HGET", key, field)

def HDel(key, *fields):
    """Delete one or more hash fields."""
    return get_client().send_command("HDEL", key, *fields)

def HGetAll(key):
    """Get all fields and values in a hash."""
    return get_client().send_command("HGETALL", key)

def HIncrBy(key, field, increment):
    """Increment the integer value of a hash field by the given amount."""
    return get_client().send_command("HINCRBY", key, field, increment)

def HExists(key, field):
    """Check if a hash field exists."""
    return get_client().send_command("HEXISTS", key, field)

def HLen(key):
    """Get the number of fields in a hash."""
    return get_client().send_command("HLEN", key)

def HKeys(key):
    """Get all field names in a hash."""
    return get_client().send_command("HKEYS", key)

def HVals(key):
    """Get all values in a hash."""
    return get_client().send_command("HVALS", key)

def HMSet(key, **mapping):
    """
    Set multiple hash fields to multiple values using keyword arguments.
    """
    args = [key]
    for k, v in mapping.items():
        args.extend([k, v])
    return get_client().send_command("HMSET", *args)

def HMGet(key, *fields):
    """Get the values of all the given hash fields."""
    return get_client().send_command("HMGET", key, *fields)

def HDelAll(key):
    """Delete all fields in a hash."""
    return get_client().send_command("HDELALL", key)

def HExpire(key, field, seconds):
    """Set expiration for a hash field in seconds."""
    return get_client().send_command("HEXPIRE", key, field, seconds)